extensions/crossmodel-lang/src/language-server/cross-model-serializer.ts

```diff
@@ -25,130 +25,140 @@ import {
+
+    serialize(root: CrossModelRoot): string {
+        if (root.entity) {
+            return this.serializeEntity(root.entity);
+        }
+        if (root.relationship) {
+            return this.serializeRelationship(root.relationship);
+        }
+        if (root.diagram) {
+            return this.serializeDiagram(root.diagram);
+        }
+        return '';
+    }
+
+    protected serializeEntity(entity: Entity): string {
+        return `entity ${entity.name} {
+    description := "${entity.description}";
+    attributes {
+${this.serializeAttributes(entity.attributes as unknown as Record<string, Attribute>)}
+    }
+}`;
+    }
 
-   private serializeAttributes(attributes: Attribute[] | undefined): string {
-      return attributes && Array.isArray(attributes)
-         ? attributes.map(attribute => `${attribute.name} := ${attribute.value}`).join(';\n')
-         : '';
-   }
+    private serializeAttributes(attributes: Record<string, Attribute> | undefined): string {
+        let result = '';
 
-   protected serializeRelationship(relationship: Relationship): string {
-      return `relationship ${relationship.name} {
+        for (const key in attributes) {
+            if (attributes.hasOwnProperty.call(attributes, key)) {
+                const attribute = attributes[key];
+                result += `\t\t${attribute.name} := '${attribute.value}';\n`;
+            }
+        }
+        result += '';
+
+        return result;
+    }
+
+    protected serializeRelationship(relationship: Relationship): string {
+        return `relationship ${relationship.name} {
          source := ${this.serializeReference(relationship.source)}${
-         relationship.sourceAttribute ? ' with ' + this.serializeReference(relationship.sourceAttribute) : ''
-      };
+            relationship.sourceAttribute ? ' with ' + this.serializeReference(relationship.sourceAttribute) : ''
+        };
          target := ${this.serializeReference(relationship.target)}${
-         relationship.targetAttribute ? ' with ' + this.serializeReference(relationship.targetAttribute) : ''
-      };
+            relationship.targetAttribute ? ' with ' + this.serializeReference(relationship.targetAttribute) : ''
+        };
          type := ${relationship.type};
          properties {
             ${this.serializeProperties(relationship.properties)}
          }
       }`;
-   }
+    }
 
-   private serializeProperties(properties: Property[] | undefined): string {
-      return properties && Array.isArray(properties) ? properties.map(property => `${property.key} := ${property.value}`).join(';\n') : '';
-   }
+    private serializeProperties(properties: Property[] | undefined): string {
+        return properties && Array.isArray(properties)
+            ? properties.map(property => `${property.key} := ${property.value}`).join(';\n')
+            : '';
```

Comment:
This serializeProperties is only used by us right? It's our own function.

I think returning '' is best for the end-user and returning an exception/error is best for debugging ourselves. So both option are valid I think.

Check whether the function is called correctly can also be part of unit testing of our application. For now I would leave it, since Ozcan didn't actually change the function.